siliconcompiler/client.py

```diff
@@ -62,8 +62,13 @@ def remote_preprocess(chip):
             chip.set('arg', 'index', index)
             func = chip.find_function(tool, 'tool', 'setup_tool')
             func(chip)
-            # Run the actual import step locally.
-            chip._runtask(local_step, index, active, error)
+
+        # Need to override steplist here to make sure check_manifest() doesn't
+        # check steps that haven't been setup.
+        chip.set('steplist', local_step)
+
+        # Run the actual import step locally.
+        chip._runtask(local_step, index, active, error)
```

Comment:
Do these steps need to be indented under 'if tool:' to avoid calling them for builtin functions?

Reply:
Now that I think about it, this may be moot since we're unlikely to have multiple pre-process/import steps. We originally wrote this to possibly support a whole pre-remote flowgraph, but it seems like we're settling on running one 'import' task.

Reply:
This was actually originally indented under the 'if tool:', but I purposely de-indented them to make sure they'd run regardless. I think even if the tool were a built-in, we would want to run it -- this seemed like a latent bug in the original code.

For example, one-shot tool runs use the 'join' builtin as a no-op import step that collects required files. If we wanted to eventually allow users to just single-step a tool on the server and we didn't call a local _runtask() for the join, I don't think SC would collect the input files and have them packaged up and sent to the server.

Reply:
Ah, that makes sense. I think the code used to call `setup_tool` directly, which was why it was like that, but the `_runstep` method should handle everything correctly.

Nice catch!